src/Microsoft.IdentityModel.Tokens/RsaCryptoServiceProviderProxy.cs

```diff
@@ -52,24 +56,20 @@ public RSACryptoServiceProviderProxy(RSACryptoServiceProvider rsa)
                 throw LogHelper.LogArgumentNullException("rsa");
 
             //
-            // If the provider does not understand SHA256, 
-            // replace it with one that does.
-            //
-            if (rsa.CspKeyContainerInfo.ProviderType != PROV_RSA_AES)
+            // Level up the provider type only if:
+            // 1. it is PROV_RSA_FULL or PROV_RSA_SCHANNEL which denote CSPs that only understand Sha1 algorithms
+            // 2. it is not associated with a hardware key
+            if ((rsa.CspKeyContainerInfo.ProviderType == PROV_RSA_FULL || rsa.CspKeyContainerInfo.ProviderType == PROV_RSA_SCHANNEL) && !rsa.CspKeyContainerInfo.HardwareDevice)
```

Comment:
We agreed earlier to not support Sha1 even for validation, I have sent you an email. Currently, when creating signature providers (either for signing or validation), we throw "Algorithm not supported" exception for sha1.